src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs

```diff
@@ -300,9 +307,17 @@ private unsafe bool TryEncryptOrDecrypt(SafeNCryptKeyHandle key, ReadOnlySpan<by
             bool encrypt,
             out int bytesNeeded)
         {
-            return encrypt ?
+            ErrorCode errorCode = encrypt ?
                 Interop.NCrypt.NCryptEncrypt(key, input, input.Length, paddingInfo, output, output.Length, out bytesNeeded, paddingMode) :
                 Interop.NCrypt.NCryptDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out bytesNeeded, paddingMode);
+
+            // Windows 10.1903 can return success when it meant NTE_BUFFER_TOO_SMALL.
+            if (errorCode == ErrorCode.ERROR_SUCCESS && bytesNeeded > output.Length)
+            {
```

Comment:
Wondering if we should clear here instead of per callsite

Reply:
In the Span one it's up to the owner to manage their data.  Only the one where we're creating arrays for you are we responsible for clearing the arrays that are being abandoned to the GC.